modules/afsocket/afinet-dest.c

```diff
@@ -697,6 +705,31 @@ afinet_dd_free(LogPipe *s)
   afsocket_dd_free(s);
 }
 
+static void
+afinet_dd_update_tls_verifier(AFSocketDestDriver *s, ReloadStoreItem *rsi)
+{
+  AFInetDestDriver *self = (AFInetDestDriver *) s;
+
+  LogWriter *writer = rsi->writer;
+
+  if (!writer)
+    return;
+
+  LogProtoClient *proto = log_writer_get_proto(writer);
+
+  if (!proto)
+    return;
+
+  LogTransport *transport = log_transport_stack_get_transport(&proto->transport_stack, LOG_TRANSPORT_TLS);
+
+  if (transport)
+    {
+      TLSSession *session = log_tansport_tls_get_session(transport);
+      AFInetDestDriverTLSVerifyData *verify_data = _get_tls_verify_data (session->verifier);
+      verify_data->signal_connector = self->super.super.super.super.signal_slot_connector;
```

Comment:
I was just wondering whether we could just drop the connection if the handshake wasn't completed yet instead of trying to keep the connection alive.

This is more like a discussion rather than a request to change.

Reply:
I'm guessing you are suggesting this because the implementation goes through a lot of layers, which looks ugly.

I think querying the transport handshake progress belongs to the `LogTransport` class, so querying such information would also go through the writer->proto->transport stack->transport chain. Because of this, I would rather keep the connection alive if `keep-alive(yes)` is set.

Reply:
We discussed everything IRL, we keep the current behavior and improve the code around ReloadStoreItem a little more.